src/py/flwr/server/fleet_event_log_interceptor.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower Fleet API event log interceptor."""
+
+
+from typing import Any, Callable, Union, cast
+
+import grpc
+
+from flwr.common.event_log_plugin import EventLogRequest, EventLogResponse
+from flwr.common.event_log_plugin.event_log_plugin import EventLogWriterPlugin
+from flwr.common.typing import LogEntry
+
+
+class FleetEventLogInterceptor(grpc.ServerInterceptor):  # type: ignore
+    """Fleet API interceptor for logging events."""
+
+    def __init__(self, log_plugin: EventLogWriterPlugin) -> None:
+        self.log_plugin = log_plugin
+
+    def intercept_service(
+        self,
+        continuation: Callable[[Any], Any],
+        handler_call_details: grpc.HandlerCallDetails,
+    ) -> grpc.RpcMethodHandler:
+        """Flower server interceptor logging logic.
+
+        Intercept all unary-unary calls from users and log the event. Continue RPC call
+        if event logger is enabled on the SuperLink, else, terminate RPC call by setting
+        context to abort.
+        """
+        # One of the method handlers in
+        # `flwr.superexec.exec_servicer.ExecServicer`
+        method_handler: grpc.RpcMethodHandler = continuation(handler_call_details)
+        method_name: str = handler_call_details.method
+        return self._generic_event_log_unary_method_handler(method_handler, method_name)
+
+    def _generic_event_log_unary_method_handler(
+        self, method_handler: grpc.RpcMethodHandler, method_name: str
+    ) -> grpc.RpcMethodHandler:
+        def _generic_method_handler(
+            request: EventLogRequest,
+            context: grpc.ServicerContext,
+        ) -> Union[EventLogResponse, Exception]:
```

Comment:
This function won't return an exception.

```suggestion
        ) -> EventLogResponse:
```